demisto_sdk/commands/common/hook_validations/playbook.py

```diff
@@ -85,6 +87,87 @@ def is_valid_version(self):  # type: () -> bool
         """
         return self._is_valid_version()
 
+    def collect_all_inputs_in_use(self):
+        """
+
+        Returns: list of all inputs used in playbook.
+
+        """
+        with open(self.file_path, 'r') as f:
+            playbook_text = f.read()
+        all_inputs_occurrences = re.findall(r"inputs\.[\S]*", playbook_text)
+        return [input.split('.')[1] for input in all_inputs_occurrences]
+
+    def collect_all_inputs_from_inputs_section(self):
+        """
+
+        Returns: list of all inputs from 'inputs' section of playbook.
+
+        """
+        inputs: Dict = self.current_file.get('inputs', {})
+        inputs_keys = []
+        for input in inputs:
+            key = input.get('key', '')
+            if key:
+                inputs_keys.append(key)
+        return inputs_keys
+
+    @staticmethod
+    def find_missing_inputs(list_to_search_for, list_to_search_in):
+        """
+
+        Args:
+            list_to_search_for: list of values to search for.
+            list_to_search_in: list of values to search in.
+
+        Returns: all values from list_to_search_for that does not appear in list_to_search_in.
+
+        """
+        result = []
+        for input in list_to_search_for:
+            if input not in list_to_search_in:
+                result.append(input)
+        return result
+
+    @error_codes('PB118')
+    def are_all_inputs_in_use(self):  # type: () -> bool
+        """Check whether the playbook inputs are in use in any of the tasks
+
+        Return:
+            bool. if the Playbook inputs are in use.
+        """
+        all_inputs_occurrences = self.collect_all_inputs_in_use()
+        inputs_list = self.collect_all_inputs_from_inputs_section()
```

Comment:
we call it once in an outer function and then call this method which should check whether `len(input_definitions_set - used_inputs_set) == 0`. similar for the other validation

Reply:
how about `input_definitions_set == used_inputs_set`?